source/test/object.ts

```diff
@@ -78,3 +78,72 @@ test('object.hasAnyKeys', t => {
 	t.throws(() => ow({unicorn: '🦄'}, ow.object.hasAnyKeys('foo')), 'Expected object to have any key of `["foo"]`');
 	t.throws(() => ow({unicorn: '🦄'}, ow.object.hasAnyKeys('unicorn.value')), 'Expected object to have any key of `["unicorn.value"]`');
 });
+
+test('object.exactShape', t => {
+	t.notThrows(() => ow({unicorn: '🦄'}, ow.object.exactShape({
+		unicorn: ow.string
+	})));
+
+	t.notThrows(() => ow({unicorn: '🦄'}, ow.object.exactShape({
+		unicorn: ow.string
+	})));
+
+	t.notThrows(() => ow({unicorn: '🦄', rainbow: {value: '🌈'}}, ow.object.exactShape({
+		unicorn: ow.string,
+		rainbow: {
+			value: ow.string
+		}
+	})));
+
+	t.throws(() => ow({unicorn: '🦄', rainbow: {value: '🌈'}}, ow.object.exactShape({
+		unicorn: ow.string,
+		rainbow: {
+			foo: ow.string
+		}
+	})), 'Expected `rainbow.foo` to be of type `string` but received type `undefined` in object');
+
+	t.throws(() => ow({unicorn: '🦄', rainbow: '🌈'}, ow.object.exactShape({
+		unicorn: ow.string
+	})), 'Did not expect property `rainbow` to exist, got `🌈` in object');
+
+	const foo = {unicorn: '🦄', rainbow: {valid: true, value: '🌈'}};
+
+	t.throws(() => ow(foo, ow.object.exactShape({
+		unicorn: ow.string,
+		rainbow: {
+			valid: ow.boolean
+		}
+	})), 'Did not expect property `rainbow.value` to exist, got `🌈` in object `foo`');
+});
+
+test('object.partialShape', t => {
+	t.notThrows(() => ow({unicorn: '🦄'}, ow.object.partialShape({
+		unicorn: ow.string
+	})));
+
+	t.throws(() => ow({unicorn: '🦄'}, ow.object.partialShape({
+		unicorn: ow.number
+	})), 'Expected `unicorn` to be of type `number` but received type `string` in object');
```

Comment:
This one might be harder because the error is actually thrown by the number predicate. That's why I added the `in object` part at the end because that was easier.  I can always use a simple replace though.